modules/jupyter-widget/src/widget.js

```diff
@@ -119,6 +124,20 @@ export class DeckGLView extends DOMWidgetView {
     super.render();
 
     this.model.on('change:json_input', this.valueChanged.bind(this), this);
+    this.model.on('change:data_buffer', this.dataBufferChanged.bind(this), this);
+    this.dataBufferChanged();
+  }
+
+  dataBufferChanged() {
+    if (this.model.get('data_buffer')) {
+      const propsWithBinary = processDataBuffer({
+        dataBuffer: this.model.get('data_buffer'),
+        jsonProps: this.model.get('json_input')
+      });
+
+      jsonConverter.convert(propsWithBinary);
+      this.deck.setProps(jsonConverter.convertedJson);
```

Comment:
The cached conversion result should be private. Use the return value of `jsonConverter.convert`.